### PR TITLE
Refactor version formatting and add tests

### PR DESCRIPTION
**Pull request by gerhard**
Follow-up to https://github.com/dagger/dagger/pull/2507#discussion_r882932740

### CODE REVIEW

**Comment by gerhard** [comment]
I didn't want to add `ginkgo ./version/` to the `Makefile`, and after 15 minutes wrangling with `ci.cue`, I just stopped.

Rain check:

image

File: version/version_test.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	"fmt"
+	"runtime"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"go.dagger.io/dagger/version"
+)
+
+var _ = Describe("Version", func() {
```
Comment (aluzzardi) [inline version/version_test.go:13]: We use `stretchr/testify` across the code base (see `context_test.go`, `git_test.go`, etc)

**Review by aluzzardi — COMMENTED** [review]
LGTM, one comment regarding tests

File: version/version_test.go (hunk 1 above)
Comment (gerhard) [reply version/version_test.go:13]: OK, I will switch to this one.

**Review by TomChv — APPROVED** [review]
LGTM

**Pull request merged**